### PR TITLE
Make h5grove compatible with h5py 2.9

### LINKED ISSUE #74: AttributeError: 'AttributeManager' object has no attribute 'get_id'
Got that error in Daiquiri CI.

Probably because there was a pbm with a file, or the file was missing.

Maybe there is a way to have better feedback.

https://gitlab.esrf.fr/ui/daiquiri/-/issues/79

```
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/daiquiri/core/components/h5grove.py", line 136, in get
    return super().get(**kwargs)
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/flask_apispec/annotations.py", line 122, in wrapped
    return wrapper(*args, **kwargs)
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/flask_apispec/wrapper.py", line 29, in __call__
    response = self.call_view(*args, **kwargs)
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/flask_apispec/wrapper.py", line 52, in call_view
    return self.func(*args, **kwargs)
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/daiquiri/core/components/h5grove.py", line 66, in get
    content = self._get_content(h5file, **kwargs)
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/daiquiri/core/components/h5grove.py", line 139, in _get_content
    return create_content(h5file, kwargs["path"]).metadata()
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/h5grove/content.py", line 228, in metadata
    *super().metadata().items(),
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/h5grove/content.py", line 132, in metadata
    for k in attribute_names
  File "/opt/conda/envs/apienv/lib/python3.7/site-packages/h5grove/content.py", line 132, in <listcomp>
    for k in attribute_names
AttributeError: 'AttributeManager' object has no attribute 'get_id'
```

### PR DESCRIPTION
**Pull request by loichuder**
Fix #74

### CODE REVIEW

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -48,6 +48,13 @@ jobs:
       - name: pytest benchmark 📈
         run: invoke benchmark
 
+      - name: pytest tests h5py==2.9 🕶️
+        run: |
+          pip install h5py==2.9
+          pushd test/
+          invoke test
+          popd
```
Comment (t20100) [inline .github/workflows/lint-test.yml:56]: :+1:

File: h5grove/utils.py [hunk 2]
```diff
@@ -18,10 +18,16 @@ class LinkError(NotFoundError):
     pass
 
 
-def attr_metadata(attrId: h5py.h5a.AttrID) -> dict:
+def get_attr_id(entity_attrs: h5py.AttributeManager, attr_name: str):
+    return h5py.h5a.open(entity_attrs._id, entity_attrs._e(attr_name))
```
[the remark this answers is not in the document]
Comment (loichuder) [reply h5grove/utils.py:22]: Yeah, that what I had at first but I found it unnecessarily complicated as this method worked for 2.x and 3.x.

But you are right, h5py internals might change in the future so it is better to call the private API only for 2.x.

File: h5grove/utils.py [hunk 3]
```diff
@@ -18,10 +19,27 @@ class LinkError(NotFoundError):
...
+
+
+def _legacy_get_attr_id(entity_attrs: h5py.AttributeManager, attr_name: str):
+    return h5py.h5a.open(entity_attrs._id, entity_attrs._e(attr_name))
+
+
+get_attr_id = (
+    _legacy_get_attr_id
+    if h5py_version.major <= 2 and h5py_version.minor <= 9
+    else _get_attr_id
```
Comment (loichuder) [inline h5grove/utils.py:33]: `get_id` was introduced in h5py 2.10: https://docs.h5py.org/en/stable/whatsnew/2.10.html

**Review by t20100 — APPROVED** [review]
LGTM

**Comment by loichuder** [comment]
Thanks for the review :slightly_smiling_face:

**Pull request merged**